tests/cplusplus/Common/RegStringMap.cpp

```diff
@@ -0,0 +1,153 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/Common/RegisteredStringMap.h>
+
+// Library/third-party includes
+#include "gtest/gtest.h"
+
+// Standard includes
+#include <string>
+
+using osvr::util::StringID;
+using osvr::util::PeerStringID;
+using osvr::common::RegisteredStringMap;
+using osvr::common::CorrelatedStringMap;
+
+class RegisteredStringMapTest : public ::testing::Test {
+  public:
+    virtual void SetUp() {
+        regID0 = regMap.getStringID("RegVal0");
+        regID1 = regMap.getStringID("RegVal1");
+        regID2 = regMap.getStringID("RegVal2");
+
+        corID0 = corMap.getStringID("CorVal0");
+        corID1 = corMap.getStringID("CorVal1");
+        corID2 = corMap.getStringID("CorVal2");
+    }
+
+    RegisteredStringMap regMap;
+    CorrelatedStringMap corMap;
+
+    StringID regID0, regID1, regID2;
+    StringID corID0, corID1, corID2;
+};
+
+TEST(RegisteredStringMap, create) {
+
+    RegisteredStringMap regMap;
+    ASSERT_EQ(0, regMap.getEntries().size());
+    ASSERT_FALSE(regMap.isModified());
+}
+
+TEST(RegisteredStringMap, addNewValues) {
+    RegisteredStringMap regMap;
+    CorrelatedStringMap corMap;
+
+    StringID regID0 = regMap.getStringID("RegVal0");
+    ASSERT_TRUE(regMap.isModified());
+    ASSERT_EQ(1, regMap.getEntries().size());
+
+    StringID regID1 = regMap.getStringID("RegVal1");
+    ASSERT_TRUE(regMap.isModified());
+    ASSERT_EQ(2, regMap.getEntries().size());
+
+    StringID regID2 = regMap.getStringID("RegVal2");
+    ASSERT_TRUE(regMap.isModified());
+    ASSERT_EQ(3, regMap.getEntries().size());
+}
+
+TEST_F(RegisteredStringMapTest, checkExistingValues) {
+
+    regMap.clearModifiedFlag();
+
+    StringID regID = regMap.getStringID("RegVal0");
+    ASSERT_FALSE(regMap.isModified());
+    ASSERT_EQ(3, regMap.getEntries().size());
+    ASSERT_EQ(0, regID.value());
+
+    regID = regMap.getStringID("RegVal1");
+    ASSERT_FALSE(regMap.isModified());
+    ASSERT_EQ(3, regMap.getEntries().size());
+    ASSERT_EQ(1, regID.value());
+
+    regID = regMap.getStringID("RegVal2");
+    ASSERT_FALSE(regMap.isModified());
+    ASSERT_EQ(3, regMap.getEntries().size());
+    ASSERT_EQ(2, regID.value());
+}
+
+TEST_F(RegisteredStringMapTest, getValues) {
+
+    ASSERT_STREQ("RegVal0", regMap.getStringFromId(regID0).c_str());
+    ASSERT_STREQ("RegVal1", regMap.getStringFromId(regID1).c_str());
+    ASSERT_STREQ("RegVal2", regMap.getStringFromId(regID2).c_str());
```

Comment:
Yes, `ASSERT_STREQ` is only for c style strings. I didn't know that `ASSERT_EQ` would work for c++ strings, but it does so those are changed.